### PR TITLE
chore: add blocklet meta spec

### PR DESCRIPTION
**Pull request by wangshijun**
- [x] Blocklet meta spec
- [x] Blocklet bundle spec

### CODE REVIEW

**Comment by wangshijun** [comment]
https://github.com/ArcBlock/abt-node/issues/1575

File: docs/meta.md [hunk 1]
```diff
@@ -12,60 +12,60 @@ This document describes how to define a blocklet that can be find/installed/mana
...
+| repository                     | Object   | No        | Defines blocklet source location                          |                                                                                                    | Draft  |
+| repository.type                | String   | No        | Repository type, can be `git`                             |                                                                                                    | Draft  |
+| repository.url                 | String   | No        | Repository URL                                            |                                                                                                    | Draft  |
+|                                |          |           |                                                           |                                                                                                    | Draft  |
+| charging                       | Object   | No        | A place holder for charging related                       |                                                                                                    | Draft  |
```
Comment (mave99a) [inline docs/meta.md]: charging --> paid

@robroyhobbs  can you also help to review those English words ?

File: docs/meta.md [hunk 2]
```diff
@@ -12,60 +12,60 @@ This document describes how to define a blocklet that can be find/installed/mana
...
+| charging.price                 | Number   | No        | The price of the blocklet, default to 0                   |                                                                                                    | Draft  |
+|                                |          |           |                                                           |                                                                                                    | Draft  |
+| stats                          | Object   | No        | A place holder for statistics related                     |                                                                                                    | Draft  |
```
Comment (mave99a) [inline docs/meta.md]: stats maybe should store elsewhere?

File: docs/meta.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+| name                           | String   | Yes       | The name of the blocklet                                  | Usually in snake case                                                                              | Draft  |
+| title                          | String   | No        | The human readable name of the blocklet                   | Defaults to title if not specified                                                                 | Draft  |
+| description                    | String   | Yes       | The description of the blocklet                           | Displayed in marketplace                                                                           | Draft  |
+| version                        | String   | Yes       | The version of the blocklet                               | Should be versioned according to [semver](https://semver.org/)                                     | Draft  |
+| group                          | Enum     | Yes       | The type of the blocklet, can be `dapp/static`            | Critical for ABT Node to decide how to mange the blocklet                                          | Draft  |
```
Comment (mave99a) [inline docs/meta.md]: It called "group" but internally it's also "type".

Comment (mave99a) [reply docs/meta.md]: We could add a category for future use.

File: docs/meta.md (hunk 1 above)
Comment (robroyhobbs) [reply docs/meta.md]: okay, no problem.

**Comment by mave99a** [comment]
可以参考 vscode 一些文档的说明格式： 

https://github.com/Microsoft/vscode-docs/blob/master/docs/remote/devcontainerjson-reference.md

File: docs/meta.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+> For integrity generating and verifying, we can leverage the [ssri](https://www.npmjs.com/package/ssri) package.
+
+## Blocklet DID Generation
+
+1. Take `name` field from blocklet meta
+2. Create buffer from name: `Buffer.from(name, 'utf8')`
+3. Use buffer as the public-key to generate the did: `fromPublicKey(buffer, { role: types.RoleType.Any })
+
+## Blocklet Bundle Format
```
Comment (mave99a) [inline docs/meta.md]: Let's move the bundle format to a separate document.  Many developers may not care about how bundle is made, but will need to know blocklet.yml format

File: docs/meta.md (hunk 2 above)
Comment (wangshijun) [reply docs/meta.md]: moved to blocklet registry extra part

**Pull request merged**